src/main/java/uk/gov/companieshouse/api/accounts/validation/DebtorsValidator.java

```diff
@@ -0,0 +1,186 @@
+package uk.gov.companieshouse.api.accounts.validation;
+
+import java.util.Optional;
+import javax.validation.Valid;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import uk.gov.companieshouse.api.accounts.exception.DataException;
+import uk.gov.companieshouse.api.accounts.exception.ServiceException;
+import uk.gov.companieshouse.api.accounts.model.rest.notes.Debtors.Debtors;
+import uk.gov.companieshouse.api.accounts.model.validation.Errors;
+import uk.gov.companieshouse.api.accounts.service.CompanyService;
+import uk.gov.companieshouse.api.accounts.transaction.Transaction;
+import uk.gov.companieshouse.api.model.company.CompanyProfileApi;
+
+@Component
+public class DebtorsValidator extends BaseValidator {
+
+    @Value("${invalid.note}")
+    private String invalidNote;
+
+    @Value("${inconsistent.data}")
+    private String inconsistentData;
+
+    private static final String DEBTORS_PATH = "$.debtors";
+    private static final String DEBTORS_PATH_PREVIOUS = DEBTORS_PATH + ".previous_period";
+    private static final String CURRENT_TOTAL_PATH = DEBTORS_PATH + ".current_period.total";
+    private static final String PREVIOUS_TOTAL_PATH = DEBTORS_PATH_PREVIOUS + ".total";
+    private static final String PREVIOUS_TRADE_DEBTORS = DEBTORS_PATH_PREVIOUS + ".trade_debtors";
+    private static final String PREVIOUS_PREPAYMENTS = DEBTORS_PATH_PREVIOUS + ".prepayments_and_accrued_income";
+    private static final String PREVIOUS_OTHER_DEBTORS = DEBTORS_PATH_PREVIOUS + ".other_debtors";
+    private static final String PREVIOUS_GREATER_THAN_ONE_YEAR = DEBTORS_PATH_PREVIOUS + ".greater_than_one_year";
+
+    private CompanyService companyService;
+
+    @Autowired
+    public DebtorsValidator(CompanyService companyService) {
+        this.companyService = companyService;
+    }
+
+    public Errors validateDebtors(@Valid Debtors debtors,
+                                  Transaction transaction) throws DataException {
+
+        Errors errors = new Errors();
+
+        if (debtors != null) {
+
+            if (debtors.getCurrentPeriod() != null) {
+
+                validateCurrentPeriodDebtors(errors, debtors);
+            }
+
+            if (debtors.getPreviousPeriod() != null) {
+
+                if (isMultipleYearFiler(transaction)) {
```

Comment:
Should we be performing the same logic for the balance sheet? If so, it may be worth abstracting this logic to the CompanyProfileService so that it can be called from multiple places.